File: practice/algorithms/greedy/priyanka_and_toys/priyanka_and_toys.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int n;
    cin >> n;
    
    vector<int> arr(n);
    for (int pos = 0; pos < n; ++pos)
    {
        cin >> arr[pos];
    }
    sort(arr.begin(), arr.end());
    
    int cnt = 0;
    int res = 0;
    while (cnt < n)
    {
        int in_cnt = 0;
        res += 1;
        while (cnt + in_cnt < n && arr[cnt + in_cnt] - arr[cnt] <= 4)
        {
            in_cnt++;
        }
        cnt += in_cnt;
    }
    
    cout << res << endl;
    
    return 0;
}

File: practice/algorithms/greedy/priyanka_and_toys/priyanka_and_toys.py
#!/usr/bin/env python

n = int(raw_input().strip())
w = map(int, raw_input().strip().split())
w.sort()

cnt = 0
res = 0

while cnt < n:
    in_cnt = 0
    res += 1
    while cnt + in_cnt < n and w[cnt + in_cnt] - w[cnt] <= 4:
        in_cnt += 1
    cnt += in_cnt
print res

